packages/config-plugins/src/ios/Google.ts

```diff
@@ -22,8 +24,40 @@ export const withGoogleServicesFile: ConfigPlugin = config => {
   });
 };
 
-export function getGoogleSignInReservedClientId(config: Pick<ExpoConfig, 'ios'>) {
-  return config.ios?.config?.googleSignIn?.reservedClientId ?? null;
+function readGoogleServicesInfoPlist(
+  relativePath: string,
+  { introspect, projectRoot }: { introspect: boolean; projectRoot: string }
+) {
+  try {
+    const googleServiceFilePath = path.resolve(projectRoot, relativePath);
+    const contents = fs.readFileSync(googleServiceFilePath, 'utf8');
+    assert(contents, 'GoogleService-Info.plist is empty');
+    return plist.parse(contents);
+  } catch (error) {
+    if (introspect) {
```

Comment:
Not sure if this is needed, if a value is defined and not found then maybe it should warn/throw and return empty. Skipping on introspect is mostly to cover the cases where a file would exist from a template, for example if you read the Info.plist or entitlements file. The Google Services file should always exist since it isn't part of the template cloning step.